flow/probes/ebpf.go

```diff
@@ -80,298 +71,24 @@ type EBPFProbesHandler struct {
-			}
-			/* disabled for now as no payload is sent
-			p := gopacket.NewPacket(C.GoBytes(unsafe.Pointer(&kernFlow.payload[0]), C.PAYLOAD_LENGTH), layers.LayerTypeTCP, gopacket.DecodeOptions{})
-			if p.Layer(gopacket.LayerTypeDecodeFailure) == nil {
-				path, app := flow.LayersPath(p.Layers())
-				f.LayersPath += "/" + path
-				f.Application = app
-			}
-			*/
-		}
-	}
-
-	// ICMP
-	if layersInfo&uint8(C.ICMP_LAYER_INFO) > 0 {
-		kind := uint8(kernFlow.icmp_layer.kind)
-		code := uint8(kernFlow.icmp_layer.code)
-		id := uint16(kernFlow.icmp_layer.id)
-
-		f.ICMP = &flow.ICMPLayer{
-			Code: uint32(code),
-			ID:   uint32(id),
-		}
-		if f.Network != nil && f.Network.Protocol == flow.FlowProtocol_IPV4 {
-			f.ICMP.Type = flow.ICMPv4TypeToFlowICMPType(kind)
-		} else {
-			f.ICMP.Type = flow.ICMPv6TypeToFlowICMPType(kind)
-		}
-	}
-
-	appLayers := strings.Split(f.LayersPath, "/")
-	f.Application = appLayers[len(appLayers)-1]
-
-	f.Metric = &flow.FlowMetric{
-		ABBytes:   int64(kernFlow.metrics.ab_bytes),
-		ABPackets: int64(kernFlow.metrics.ab_packets),
-		BABytes:   int64(kernFlow.metrics.ba_bytes),
-		BAPackets: int64(kernFlow.metrics.ba_packets),
-		Start:     f.Start,
-		Last:      f.Last,
-	}
-
-	key := kernFlowKey(kernFlow)
-
-	f.UpdateUUID(key, flow.Opts{LayerKeyMode: flow.L3PreferedKeyMode})
-
-	fops = append(fops, &flow.Operation{
-		Type: flow.ReplaceOperation,
-		Flow: f,
-		Key:  key,
-	})
-	return fops
-}
-
-func (p *EBPFProbe) updateFlowOperation(ebpfFlow *EBPFFlow, kernFlow *C.struct_flow, startKTimeNs int64, start time.Time) *flow.Operation {
-	f := flow.NewFlow()
-	f.Last = common.UnixMillis(ebpfFlow.last)
-	layersInfo := uint8(kernFlow.layers_info)
-	if layersInfo&uint8(C.TRANSPORT_LAYER_INFO) > 0 {
-		protocol := uint8(kernFlow.transport_layer.protocol)
-		switch protocol {
-		case syscall.IPPROTO_TCP:
-			f.TCPMetric = &flow.TCPMetric{
-				ABSynStart: tcpFlagTime(kernFlow.transport_layer.ab_syn, startKTimeNs, start),
-				BASynStart: tcpFlagTime(kernFlow.transport_layer.ba_syn, startKTimeNs, start),
-				ABFinStart: tcpFlagTime(kernFlow.transport_layer.ab_fin, startKTimeNs, start),
-				BAFinStart: tcpFlagTime(kernFlow.transport_layer.ba_fin, startKTimeNs, start),
-				ABRstStart: tcpFlagTime(kernFlow.transport_layer.ab_rst, startKTimeNs, start),
-				BARstStart: tcpFlagTime(kernFlow.transport_layer.ba_rst, startKTimeNs, start),
-			}
-		}
-	}
-	f.Metric = &flow.FlowMetric{
-		ABBytes:   int64(kernFlow.metrics.ab_bytes),
-		ABPackets: int64(kernFlow.metrics.ab_packets),
-		BABytes:   int64(kernFlow.metrics.ba_bytes),
-		BAPackets: int64(kernFlow.metrics.ba_packets),
-		Start:     f.Start,
-		Last:      f.Last,
-	}
-
-	key := kernFlowKey(kernFlow)
-
-	return &flow.Operation{
-		Type: flow.UpdateOperation,
-		Flow: f,
-		Key:  key,
-	}
-}
-
 func (p *EBPFProbe) run() {
 	var info syscall.Sysinfo_t
 	syscall.Sysinfo(&info)
 
-	_, flowChanOperation := p.flowTable.Start()
+	_, flowChanOperation, _ := p.flowTable.Start()
```

Comment:
please change flowChanOperation to flowEBPFChan

Reply:
done